custom_components/candy/binary_sensor.py

```diff
@@ -0,0 +1,325 @@
+            ]
+        )
+    elif type(coordinator.data) is TumbleDryerStatus:
+        async_add_entities(
+            [
+                CandyTumbleDryerBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is OvenStatus:
+        async_add_entities(
+            [
+                CandyOvenBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is HoodStatus:
+        async_add_entities(
+            [
+                CandyHoodBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is FridgeStatus:
+        async_add_entities(
+            [
+                CandyFridgeBinarySensor(coordinator, config_id)
+            ]
+        )
+    elif type(coordinator.data) is HobStatus:
+        async_add_entities(
+            [
+                CandyHobBinarySensor(coordinator, config_id)
+            ]
+        )                
+    elif type(coordinator.data) is DishwasherStatus:
+        async_add_entities(
+            [
+                CandyDishwasherBinarySensor(coordinator, config_id)
+            ]
+        )
+    else:
+        raise Exception(f"Unable to determine machine type: {coordinator.data}")
+
+
+class CandyBaseBinarySensor(CoordinatorEntity, BinarySensorEntity):
+    def __init__(self, coordinator: DataUpdateCoordinator, config_id: str):
+        super().__init__(coordinator)
+        self.config_id = config_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.config_id)},
+            name=self.device_name(),
+            manufacturer="Candy",
+            suggested_area=self.suggested_area(),
+        )
+
+    @abstractmethod
+    def device_name(self) -> str:
+        pass
+
+    @abstractmethod
+    def suggested_area(self) -> str:
+        pass
+
+
+class CandyWashingMachineBinarySensor(CandyBaseBinarySensor):
+
+    def device_name(self) -> str:
+        return DEVICE_NAME_WASHING_MACHINE
+
+    def suggested_area(self) -> str:
+        return SUGGESTED_AREA_BATHROOM
+
+    @property
+    def name(self) -> str:
+        return self.device_name()
+
+    @property
+    def unique_id(self) -> str:
+        return UNIQUE_ID_WASHING_MACHINE_BINARY.format(self.config_id)
+
+    @property
+    def state(self) -> StateType:
+        status: WashingMachineStatus = self.coordinator.data
+        return str(status.machine_state)
+
+    @property
+    def icon(self) -> str:
+        return "mdi:washing-machine"
+
+    @property
+    def available(self):
+        status: WashingMachineStatus = self.coordinator.data
+        return status is not None
+
+    @property
+    def is_on(self):
+        status: WashingMachineStatus = self.coordinator.data
+        return status.machine_state not in [MachineState.IDLE] 
+
+
+class CandyTumbleDryerBinarySensor(CandyBaseBinarySensor):
```

Comment:
Same concern as with washing machines.

Reply:
This sensors just tell us if apliance work or not.
This information is useful in some UI integrations.

Example: I use  [minimalistic-area-card](https://github.com/junalmeida/homeassistant-minimalistic-area-card).  This card just show me white icon when appliance is on and gray icon when is of. But  homeassistant-minimalistic-area-card can use only binary sensor for this feature.